src/shared/actions.rs

```diff
@@ -17,8 +17,7 @@ pub enum DownloadAction {
 }
 
 #[derive(Clone, DefaultNone, Debug)]
-pub enum TooltipAction {
-    Show(String, DVec2),
-    Hide,
+pub enum OverlayWidgetAction {
```

Comment:
Not a very important opinion:

I fell as a user of the related widgets, would like to trigger `ModalAction::Close`, `PopupNotification::Close`, etc.

I imagine this can conflict in cases like tooltips inside modals and also, if we add an action like this to makepad, any contributor of makepad will need to use this action when they create overlay widgets to be consistent (and will need to know about it's existence). Also, it's more obvious the usage of this action if it lives near the widget it affects. So there may be no value on being DRY here if that is the intention. 

But I also feel I may be missing something, so I'm not completely sure about what I'm saying :P

Reply:
This is a good point. In fact, I was able to reproduce a situation where closing a modal also closed a popup notification, and it was not the desired behavior. Will change to `ModalAction::Close` and `PopupNotification::Close` for now, but we will need in the future a more granular way to say "close this specific modal instance".

Reply:
I actually removed everything :) I found confusing to use "Action enums" to communicate to modals/popups from the outside or from the inner contents) because those "Action enums" tends to be used to emit internal events of te widgets.

So, I think it is a safe bet for now to rely on custom actions emitted by the inner content of modals/popups and have the owner widgets of those responsible to directly invoke the `.close()` function. It is a bit more manual, but the dev has full and granular control.

See [my last commit](https://github.com/moxin-org/moxin/pull/214/commits/60bf3faf967c711363621612b233ee539b2bb69f) to check the approach.